src/Client/widgets/Task.fs

```diff
@@ -10,157 +10,226 @@ open Fable.FontAwesome
+            | Shown -> model
+            | _ -> checkAnswer' ()
+
 let init taskName getTask =
     { TaskName = taskName
-      Word = None
-      Answers = None
-      Input = ""
-      Result = None
-      AnswerShown = false },
+      State = Fetching },
       loadTaskCmd getTask
 
 let update msg model getTask =
     match msg with
-    | SetInput input ->
-        { model with Input = input }, Cmd.none
-    | SubmitAnswer -> 
-        let result = if not model.AnswerShown then 
-                        model.Answers |> Option.map (Array.contains model.Input)
-                     else model.Result
-   
-        if model.Word.IsSome && result.IsSome
-        then log model.TaskName model.Word.Value model.Input result.Value
-        
-        { model with Result = result }, Cmd.none
-    | UpdateTask ->
-        { model with Word = None; Input = ""; Result = None }, loadTaskCmd getTask
-    | ShowAnswer ->
-        let answer = model.Answers
-                    |> Option.map Array.tryHead
-                    |> Option.flatten
-                    |> Option.defaultValue ""
-        { model with Input = answer; Result = Some false; AnswerShown = true}, Cmd.none
     | FetchedTask task -> 
         { model with 
-            Word = task |> Option.map (fun t -> t.Word)
-            Answers = task |> Option.map (fun t -> t.Answers)
-            AnswerShown = false
+            State = TaskProvided { 
+                        Word = task |> Option.map (fun t -> t.Word) |> Option.defaultValue DefaultWord; 
+                        Answers = task |> Option.map (fun t -> t.Answers) |> Option.defaultValue DefaultAnswers
+                    }
         }, Cmd.none
     | FetchError _ ->
         model, Cmd.none
+    | SetAnswer input ->
+        let newState = 
+            if input <> "" then
+                InputProvided (getStateTask model.State, input, Unknown)
+            else
+                TaskProvided (getStateTask model.State)
+        { model with State = newState }, Cmd.none
+    | ShowAnswer ->
+        let task = getStateTask model.State
+        let answer = task.Answers |> Array.tryHead |> Option.defaultValue DefaultWord
+        { model with State = InputProvided (task, answer, Shown) }, Cmd.none
+    | CheckAnswer -> 
+        checkAnswer model, Cmd.none
+    | NextQuestion ->
+        { model with State = Fetching }, loadTaskCmd getTask
+
+type InputViewState = {
+    Word : ReactElement
+    InputClass : string
+    InputText : string
+}
 
-let view model dispatch =
-    let inputClass, inputIcon =
-        match model.Result with 
-        | Some result -> 
-            if result then "task-input-correct", Fa.Solid.CheckCircle  else "task-input-incorrect", Fa.Solid.TimesCircle
-        | None ->
-            "task-input-none", Fa.Solid.QuestionCircle
-
-    let task = 
-        match model.Word with
-        | Some t -> 
-            str t
-        | None ->
+let inputView model dispatch handleKeyDown =
```

Comment:
Task is now split into two sections, input view which handles text input and its states and button view which does button handling accordingly.